cluster/service/cluster.go

```diff
@@ -165,6 +165,61 @@ func Start(ctx context.Context, factory service.ClusterCacheFactory, options ...
 	return clusterCache != nil && started == uint32(1), nil
 }
 
+func (s *clusterService) LinkExistingIdentitiesToCluster(ctx context.Context, options ...rest.HTTPClientOption) error {
+	identitiesWithClusters, err := s.Repositories().Identities().GetIdentitiesWithClusterURL(ctx)
+	if err != nil {
+		return errors.Wrapf(err, "failed to load existing identities with default IDP")
+	}
+
+	if _, err := s.linkIdentitiesToCluster(ctx, identitiesWithClusters, options...); err != nil {
+		return errors.Wrapf(err, "failed while linking existing identities to it's cluster url")
+	}
+	return nil
+}
+
+func (s *clusterService) linkIdentitiesToCluster(ctx context.Context, identitiesWithClusterURL map[uuid.UUID]string, options ...rest.HTTPClientOption) (chan error, error) {
+	signer := newJWTSASigner(ctx, s.config, options...)
+	client, err := signer.createSignedClient()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to create JWT signer for cluster service")
+	}
+
+	errs := make(chan error, len(identitiesWithClusterURL))
+	go func() {
+		defer close(errs)
+		for id, u := range identitiesWithClusterURL {
+			if e := s.link(ctx, client, id, u); e != nil {
+				errs <- e
+			}
+		}
```

Comment:
Log the info message when the linking is started and when it's finished with the total count of the linked identities. Then we will know when the migrations is complete.